pepper-apis/src/main/java/org/broadinstitute/ddp/route/GetUserAnnouncementsRoute.java

```diff
@@ -57,46 +56,48 @@ public GetUserAnnouncementsRoute(I18nContentRenderer renderer) {
                 throw ResponseUtil.haltError(response, HttpStatus.SC_NOT_FOUND, err);
             }
 
-            UserDto participantDto = handle.attach(JdbiUser.class).findByUserGuid(participantGuid);
-            if (participantDto == null) {
-                ApiError err = new ApiError(ErrorCodes.USER_NOT_FOUND, "Could not find participant with guid " + participantGuid);
+            User user = handle.attach(UserDao.class).findUserByGuid(userGuid).orElse(null);
+            if (user == null) {
+                ApiError err = new ApiError(ErrorCodes.USER_NOT_FOUND, "Could not find user with guid " + userGuid);
                 LOG.warn(err.getMessage());
                 throw ResponseUtil.haltError(response, HttpStatus.SC_NOT_FOUND, err);
             }
 
             UserAnnouncementDao announcementDao = handle.attach(UserAnnouncementDao.class);
 
-            Set<Long> announcementIds = new HashSet<>();
             List<UserAnnouncement> announcements = announcementDao
-                    .findAllForParticipantAndStudy(participantDto.getUserId(), studyDto.getId())
-                    .peek(announcement -> announcementIds.add(announcement.getId()))
+                    .findAllForUserAndStudy(user.getId(), studyDto.getId())
                     .collect(Collectors.toList());
 
-            LOG.info("Found {} announcements for participant {} and study {}", announcements.size(), participantGuid, studyGuid);
+            LOG.info("Found {} announcements for user {} and study {}", announcements.size(), userGuid, studyGuid);
 
             if (!announcements.isEmpty()) {
                 try {
                     long langCodeId = handle.attach(JdbiLanguageCode.class).getLanguageCodeId(langCode);
                     renderer.bulkRenderAndApply(handle, announcements, style, langCodeId);
                 } catch (NoSuchElementException e) {
                     ApiError err = new ApiError(ErrorCodes.SERVER_ERROR, String.format(
-                            "Error while rendering announcement messages for participant %s and study %s",
-                            participantGuid, studyGuid));
+                            "Error while rendering announcement messages for user %s and study %s",
+                            userGuid, studyGuid));
                     LOG.error(err.getMessage(), e);
                     throw ResponseUtil.haltError(response, HttpStatus.SC_INTERNAL_SERVER_ERROR, err);
                 }
 
-                // Currently, announcements read are automatically deleted.
-                int numDeleted = announcementDao.deleteByIds(announcementIds);
-                if (numDeleted != announcements.size()) {
+                Set<Long> transientAnnouncementIds = announcements.stream()
+                        .filter(ann -> !ann.isPermanent())
+                        .map(UserAnnouncement::getId)
+                        .collect(Collectors.toSet());
```

Comment:
Good call on `Set` here to protect against "no such row" if something goes haywire server-side and returns redundant rows.